### PR TITLE
enforce name length limit

### PR DESCRIPTION
**Pull request by michaelkirk**
PTAL @charlesmchen

### CODE REVIEW

File: Signal/src/util/TextFieldHelper.swift [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+//
+
+import Foundation
+import UIKit
+
+@objc class TextFieldHelper: NSObject {
+
+    // Used to implement the UITextFieldDelegate method: `textField:shouldChangeCharactersInRange:replacementString` but takes advantage of
+    // Swifts superior unicode handling to append partial pasted text without splitting multi-byte characters.
+    class func textField(_ textField: UITextField, shouldChangeCharactersInRange editingRange: NSRange, replacementString: String, byteLimit: UInt) -> Bool {
```
Comment (charlesmchen) [inline Signal/src/util/TextFieldHelper.swift]: nit: comments and code have long line length making it harder to review.

File: Signal/src/util/TextFieldHelper.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+@objc class TextFieldHelper: NSObject {
+
+    // Used to implement the UITextFieldDelegate method: `textField:shouldChangeCharactersInRange:replacementString` but takes advantage of
+    // Swifts superior unicode handling to append partial pasted text without splitting multi-byte characters.
```
Comment (charlesmchen) [inline Signal/src/util/TextFieldHelper.swift]: s/Swifts/Swift's/

File: Signal/src/util/TextFieldHelper.swift [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        let lengthOfRemainingExistingString = byteLength(existingString) - byteLength(removedString)
+
+        let newLength = lengthOfRemainingExistingString + byteLength(replacementString)
+
+        if (newLength <= byteLimit) {
+            return true
+        }
+
+        // Don't allow any change if inserting a single char is already over the limit (typically this means typing)
+        if (replacementString.characters.count < 2) {
```
Comment (charlesmchen) [inline Signal/src/util/TextFieldHelper.swift]: Why special case this?

Comment (michaelkirk) [reply Signal/src/util/TextFieldHelper.swift]: Agreed that the behavior should be equivalent without this check, but this simple check for the common case (when a user is typing in their name) avoids the more complex logic/additional allocations required for the more complex case.

**Pull request merged**